### PR TITLE
Update DataGridCellInfo.cs

### PR DESCRIPTION
**Pull request by NotThatBen**
Proposed solution for #3297. Make DataGridCellInfo struct independent from row virtualization processes.

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCellInfo.cs [hunk 1]
```diff
@@ -184,7 +184,7 @@ public override bool Equals(object obj)
 
         internal bool EqualsImpl(DataGridCellInfo cell)
         {
-            return (cell._column == _column) && (cell.Owner == Owner) && (cell._info == _info);
+            return (cell._column == _column) && (cell.Owner == Owner) && (cell.Info == Info);
```
Comment (miloush) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCellInfo.cs:187]: How does this even compile, as far as I can see there is no `Info` member in the struct?

Comment (NotThatBen) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCellInfo.cs:187]: Ughh) Sorry.. There should be Item property. I'll fix.

**Comment by miloush** [comment]
Since this is in your repro I don't really mind, but I do want to point out that what you propose is a breaking change in public API and deviates from documented behavior.

Also note that apart from other issues, your proposed solution will return equality for an array of the same items while the current implementation does not.

**Comment by NotThatBen** [comment]
You are right. It's not good solution. I'm closing request.

**Pull request closed without merging**